crates/router/src/utils/user_role.rs

```diff
@@ -53,6 +55,11 @@ pub fn validate_role_groups(groups: &[PermissionGroup]) -> UserResult<()> {
             .attach_printable("Organization manage group cannot be added to role");
     }
 
+    if groups.iter().collect::<HashSet<_>>().len() != groups.len() {
```

Comment:
Minor suggestion: We can store this set and use it to check if `OrganizationManage` group is present more efficiently.

Reply:
yes, it will improve efficiency of search